internal/serve/httphandler/login_handler.go

```diff
@@ -101,6 +87,23 @@ func (h LoginHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	userRoleCanBypassReCAPTCHA := slices.Contains(user.Roles, data.APIUserRole.String())
+
+	if !h.ReCAPTCHADisabled && !userRoleCanBypassReCAPTCHA {
+		// validating reCAPTCHA Token
+		isValid, recaptchaErr := h.ReCAPTCHAValidator.IsTokenValid(ctx, reqBody.ReCAPTCHAToken)
+		if recaptchaErr != nil {
+			httperror.Unauthorized("Cannot validate reCAPTCHA token", recaptchaErr, nil).Render(rw)
+			return
+		}
+
+		if !isValid {
+			log.Ctx(ctx).Errorf("reCAPTCHA token is invalid for request with email %s", utils.TruncateString(reqBody.Email, 3))
+			httperror.BadRequest("reCAPTCHA token invalid", nil, nil).Render(rw)
+			return
+		}
+	}
+
```

Comment:
🚨 the purpose of having recaptcha validated before the authentication is to prevent bots from brute forcing their way in.

By moving this recaptcha after the authentication call, you're removing these safety guardrails from this endpoint.

Reply:
Addressed [abeb515](https://github.com/stellar/stellar-disbursement-platform-backend/pull/368/commits/abeb515c89402d855ecb6723f3e4f12c94cdd3c7)